draft-ietf-anima-brski-prm.md

```diff
@@ -405,26 +410,25 @@ To enable reuse of BRSKI defined functionality as much as possible, BRSKI-PRM:
+  The protocol used between the pledge and the Registrar-Agent is assumed to be HTTP(S) in the context of this document.
+  Any other protocol can be used as long as it supports the exchange of the necessary artifacts.
   This includes CoAP or protocol to be used over Bluetooth or NFC connections
   A pledge acting as a server during bootstrapping leads to the following differences compared to BRSKI:
   
-  * The pledge is discovered by the Registrar-Agent as defined in {#discovery_uc2_ppa}.
+  * The pledge is discovered by the Registrar-Agent as defined in {{discovery_uc2_ppa}}.
   * The pledge offers additional endpoints as defined in {{pledge_ep}}, so that the Registrar-Agent can request data required for bootstrapping the pledge.
   * The pledge includes additional data in the PVR, which is provided by the Registrar-Agent in the voucher-request trigger as defined in {{tpvr}}.
-    This allows the registrar to identify, with which Registrar-Agent the pledge was in contact.
-  * The order of exchanges in the BRSKI-PRM call flow is different from those in BRSKI {{!RFC8995}}, as the PVR and PER are collected simultaneously and provided to the registrar.
+    This allows the registrar to identify with which Registrar-Agent the pledge was in contact (see {{agt_prx}}).
+  * The order of exchanges in the BRSKI-PRM call flow is different from those in BRSKI {{!RFC8995}}, as the PVR and PER are collected simultaneously and provided to the registrar together.
     This enables bulk bootstrapping of several devices.
-  * The data objects utilized for the data exchange between the pledge and the registrar are self-contained authenticated objects (signature-wrapped objects).
+  * The artifacts exchanged between the pledge and the registrar via the Registrar-Agent are authenticated self-contained objects (i.e., signature-wrapped data objects).
 
 * Registrar-Agent: Provides a store and forward communication path to exchange data objects between the pledge and the domain registrar.
   The Registrar-Agent acts as a broker in situations in which the domain registrar is not directly reachable by the pledge.
   This may be due to a different technology stack or due to missing connectivity.
 
   * The Registrar-Agent triggers one or more pledges to create bootstrapping artifacts such as the voucher-request and the Enroll-Request.
     It can then perform a (bulk) bootstrapping based on the collected data.
-  * The Registrar-Agent is expected to possess information about the domain registrar: the registrar EE certificate, LDevID(CA) certificate, and IP address, either by configuration or by using the discovery mechanism defined in {{!RFC8995}}.
   * There is no trust assumption between the pledge and the Registrar-Agent as only authenticated self-contained objects are used, which are transported via the Registrar-Agent and provided either by the pledge or the domain registrar.
```

Comment:
Proposal to change to:
To facilitate this, the Registrar-Agent possess information about the domain registrar: the registrar EE certificate, LDevID(CA) certificate, and IP address, which is provided either by configuration or by using the discovery mechanism defined in {{!RFC8995}}.

Reply:
All these details are in Section 6.2 with the corresponding normative statements.
In my opinion, it goes way to far for the overview and it was even conflicting with the normative statements in Section 6.2 (no statement on LDevID(CA) certificate nor on IP address.

IP address context is discussed at length in 6.2.1.

Probably, we need a clear normative statement about the LDevID(CA) certificate in Section 6.2.

Reply:
Okay, maybe we can have a forward reference to the details as for the pledge. Proposal: 

Provides a store and forward communication path to exchange data objects between the pledge and the domain registrar as outlined in section 6.2.

Reply:
I tried to better link to the corresponding detail sections. However, I still do not understand why this configuration detail is so important in the overview and what benefit the reader would have knowing those at this point.

The communication is rather discussed in Section 7.
Section 5.1 is about the overview over all components involved in BRSKI-PRM and the high-level differences to BRSKI.
Section 6 has the subsections with the implementation/configuration details for changed components, i.e., also for pledge and registrar.

More important would be to clarify the mentioning of the "LDevID(CA) certificate". It is mentioned nowhere else in the document I cannot see how the Registrar-Agent would need this beforehand. It does forward wrapped CA certs as part of the exchanges, but that is a different aspect.

Reply:
In general, the overview may not contain all details about the pledge configuration, also to avoid doubling text, you are right.  So the forward looking reference makes it from my understanding. 

The Registrar-Agent needs to LDevID(CA certificate) to be able to validate the registrar certificate when connecting to it. That said, I would propose to move the following adopted sentence to section 6.2 as a replacement of the last sentence in the second paragraph:

The Registrar-Agent is possess information about the domain registrar: the registrar EE certificate, LDevID(CA) certificate, and IP address, either by configuration or by using on own {{!RFC8995}} run in which the Registrar-Agent may also receive its Registrar-Agent EE certificate.

Reply:
Yes, I think these clarifications help a lot!

It also connects to my other question posed on the chat about EE vs LDevID certificate. In particular the Agent-Proximity Assertion only works if the Registrar-Agent has an LDevID certificates opposed to an IDevID certificate. Further, the trust to the Registrar only relies on the domain CA certificate, if the Registrar is using an LDevID.

So why are there so many EE certificate statements, when effectively they have to be LDevID certificates issued by the domain owner (via domain CA)?

Reply:
Actually, the EE certificate and the LDevID certificate are more or less synonym. More or less as EE certificate essentially refers to the End Entity certificate, while the LDevID certificate refers to a locally significant device certificate. Both are provided via an issuing CA. LDevID underlines that the issuer is an entity in the target domain, while EE certificate may be more general. As we talk about enrolling for device certificates for the domain, it should be implicitly clear, that the certificates are issued by an issuing domain related or belonging to the domain.

Reply:
My problem is that the Terminology section defines "EE certificate" as:

> EE certificate:
: Either IDevID certificate or LDevID certificate of the EE.

When it is the IDevID, it has no relation to the domain and complexity for verification explodes..

Reply:
Okay, understood. The pledge only uses the IDevID, when no LDevID is available. The PVR can only be signed by the IDevID. The PER is signed with the IDevID in the inital case and would be signed with LDevID in case of re-enroll

Reply:
But again: Why use the terms "Registrar-Agent EE certificate" and "domain registrar EE certificate", which allow IDevID certificates for both components, while BRSKI-PRM only fully works with LDevID certificates for those components?

Is the intention to explicitly allow IDevID certificates and run BRSKI-PRM that way? Is this a valid way to configure the domain, i.e., without LDevID for these two components?

Reply:
Not quite, BRSKI (in all flavors) starts with IDevID certs to establish LDevID certs. The operator is typically not interested in utilizing the IDevID certs for operation, as he cannot enforce the lifecycle management for them.

Reply:
That is again what I would expect. So again: why does the draft say "EE certificate" at most places? Why does it imply IDevID cert is okay for these components (following the definition for EE in the draft)?

Why is "EE" in the draft? Can I replace it consistently with "LDevID" certificate/credentials for RegAgt and Registrar? Or does that break something?

Reply:
If you find LDevID too narrow, because there are domain certificates that do not follow the IEEE standard, I would rather clarify in the Terminology that wherever we say "LDevID", the domain may also use general domain-sigend EE certificates.

We need a distinction between domain certs and non-domain certs. Currently "EE" may be both, making it very confusing.

Reply:
After further thinking about it, it should be appropriate to state for the Registrar-Agent and the Registrar that they use an LDevID certificate as they also would not use IDevID certs for operation. What makes me say "should" is that the DevID module is a specific realization, which is essentially an X.509 certificate with some requirements to contained content and the DevID module has an API to manage the credentials. Hm, if we follow the definition from RFC 8995: 

LDevID:  A Local Device Identifier X.509 certificate installed by the owner of the equipment. This is a term from 802.1AR [[IDevID](https://1.ieee802.org/security/802-1ar)].

it should be fine to replace it.

Reply:
Great, that helps my fundamental understanding and I hope also the reader.
We should document the aspect of not requiring full .1AR compliance, for instance in the the Terminology section.